insights/parsers/zipl_conf.py

```diff
@@ -0,0 +1,113 @@
+            parameters="root=/dev/mapper/rhel_gss5-root crashkernel=auto rd.dasd=0.0.0100 rd.dasd=0.0.0101 rd.dasd=0.0.0102 rd.lvm.lv=rhel_gss5/root rd.lvm.lv=rhel_gss5/swap net.ifnames=0 rd.znet=qeth,0.0.0600,0.0.0601,0.0.0602,layer2=0,portname=gss5,portno=0"
+        # Configuration for dumping to SCSI disk
+        # Separate IPL and dump partitions
+        [dumpscsi]
+        target=/boot
+        dumptofs=/dev/sda2
+        parameters="dump_dir=/mydumps dump_compress=none dump_mode=auto"
+        # Menu containing two DASD boot configurations
+        :menu1
+        1=linux
+        2=linux-0-rescue-a27932c8d57248e390cee3798bbd3709
+        default=1
+        prompt=1
+        timeout=30
+
+    Examples:
+        >>> zipl_info['linux']['image']
+        '/boot/vmlinuz-3.10.0-693.el7.s390x'
+        >>> zipl_info.images
+        {'linux':'/boot/vmlinuz-3.10.0-693.el7.s390x','linux-0-rescue-a27932c8d57248e390cee3798bbd3709':'/boot/vmlinuz-0-rescue-a27932c8d57248e390cee3798bbd3709'}
+        >>> zipl_info.dumptofses
+        {'dumpscsi':'/dev/sda2'}
+        >>> zipl_info[':menu1']['1']
+        'linux'
+        >>> 'defaultauto' in zipl_info['global']
+        True
+        >>> zipl_info['global']['defaultauto']
+        None
+    """
+    def __init__(self, *args, **kwargs):
+        self._images = {}
+        self._dumptofses = {}
+        super(ZiplConf, self).__init__(*args, **kwargs)
+
+    def parse_content(self, content):
+        DEFLSECT = "defaultboot"
+        self.data = {DEFLSECT: {}}
+        current = DEFLSECT
+        for line in get_active_lines(content):
+            if line.startswith('[') and line.endswith(']'):
+                current = line[1:-1]
+                self.data[current] = {}
+            elif line.startswith(':'):
+                current = line
+                self.data[current] = {}
+            else:
+                if '=' in line:
+                    k, v = [s.strip() for s in line.split('=', 1)]
+                    if k == "image":
+                        self._images[current] = v
+                    elif k == "dumptofs":
+                        self._dumptofses[current] = v
+                    self.data[current][k] = v
+                else:
+                    self.data[current][line] = True
+
+    @property
+    def images(self):
+        """
+        Get all `image` items referenced in zipl configuration file
+
+        Returns:
+            (dict): Returns a dict of the `section` and `image` names referenced
+                    in zipl configuration file
+        """
+        return self._images
+
+    @property
+    def dumptofses(self):
+        """
+        Get all `dumptofs` items referenced in zipl configuration file
+
+        Returns:
+            (dict): Returns a dict of the `section` and `dumptofs` names referenced
+                    in zipl configuration file
+        """
+        return self._dumptofses
```

Comment:
@lonicerae why not just use the attributes `self.images` and self.dumptofses`?

Reply:
Hi @bfahr , because we want them read only.